pkg/query/flamegraph_arrow.go

```diff
@@ -942,19 +959,36 @@ func (fb *flamegraphBuilder) NewRecord() (arrow.Record, error) {
 		fb.builderCumulative.NewArray(),
 		fb.builderDiff.NewArray(),
 	}
+	defer func() {
+		for i, arr := range arrays {
+			switch i {
+			case 4, 5, 10, 11, 12:
+				// These arrays have already been defered release above where they were created.
```

Comment:
this seems pretty brittle, maybe let's rather spell them out?

Reply:
Moved all array creation to append to a single slice that gets cleaned up at the end.